src/operun/crm/ldap/ldap_sync.py

```diff
@@ -137,10 +172,63 @@ def unbind(self):
         try:
             self.connection.unbind()
         except AttributeError:
-            logger.info(_('No connection!'))
+            pass
         else:
             logger.info(_('Disconnected from LDAP server...'))
 
+    def search_for_user(self, obj):
+        """
+        Search for user by UID and return DN from LDAP.
+        Can be used as conditional method if user exists in LDAP.
+        Should take single object, either passed or through iteration.
+        """
+        connection = self.connect()
+        ldap_objectclass_mapping = api.portal.get_registry_record(
+            name='operun.crm.ldap_objectclass_mapping')
+        accounts_dn = api.portal.get_registry_record(
+            name='operun.crm.accounts_dn')
+        users_dn = api.portal.get_registry_record(name='operun.crm.users_dn')
+        # Defaults
+        content_type = obj.Type()
+        object_class = self.list_to_dict(
+            ldap_objectclass_mapping)[content_type]
+        # Check Type
+        if content_type == 'Account':
+            ldap_dn = accounts_dn
+        if content_type == 'Contact':
+            ldap_dn = users_dn
+        try:
+            # Returns DN(s) if user exists in LDAP
+            search_result = connection.search_s(ldap_dn, ldap.SCOPE_SUBTREE, '(&(uid={0})(objectClass={1}))'.format(obj.UID(), object_class))  # noqa
+        except ldap.LDAPError:
+            pass
+        else:
+            return search_result
+        self.unbind()
+
+    def update_node_tree(self, obj):
```

Comment:
This needs some explanation. We should not delete duplicate objects. Need to find a better solution.